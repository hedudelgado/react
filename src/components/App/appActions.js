import { HeaderInterface, InvalidSessionErrorCodeMap, InvalidSessionHelper } from 'mega-code';
import {
  getCountries,
  getNewsletterRegions,
  getNewsletterPreferences,
  getProfile,
  updateProfile,
  deleteCustomer,
} from '../../utils/api';

// actions
export const USER_LOGGED_IN_SUCCESSFULLY = 'USER_LOGGED_IN_SUCCESSFULLY';
export const COUNTRIES_RECEIVED = 'COUNTRIES_RECEIVED';
export const PROFILE_RECEIVED = 'PROFILE_RECEIVED';
export const REGIONS_RECEIVED = 'REGIONS_RECEIVED';
export const NEWSLETTER_RECEIVED = 'NEWSLETTER_RECEIVED';
export const SET_EDIT_VIEW = 'SET_EDIT_VIEW';
export const SET_NOTIFICATION_MESSAGE = 'SET_NOTIFICATION_MESSAGE';
export const PROFILE_UPDATED = 'PROFILE_UPDATED';
export const NEWSLETTER_UPDATED = 'NEWSLETTER_UPDATED';
export const RESET_NEWSLETTER_INFO = 'RESET_NEWSLETTER_INFO';
export const RESET_USER_INFO = 'RESET_USER_INFO';
// views
export const DEFAULT_VIEW = undefined;
export const USER_DETAILS_VIEW = 'USER_DETAILS_VIEW';
export const PASSWORD_VIEW = 'PASSWORD_VIEW';
export const PAYMENT_CARD_VIEW = 'PAYMENT_CARD_VIEW';
export const ADD_GUEST_VIEW = 'ADD_GUEST_VIEW';
export const REGULAR_GUESTS_VIEW = 'REGULAR_GUESTS_VIEW';
export const NEWSLETTER_VIEW = 'NEWSLETTER_VIEW';
export const ROOM_REQUIREMENTS_VIEW = 'ROOM_REQUIREMENTS_VIEW';
export const EXTRAS_PREFERENCES_VIEW = 'EXTRAS_PREFERENCES_VIEW';
export const MEMORABLE_WORD_VIEW = 'MEMORABLE_WORD_VIEW';
// errors
export const PROFILE_ERROR = 'PROFILE_ERROR';
export const PASSWORD_ERROR = 'PASSWORD_ERROR';
export const NEWSLETTER_ERROR = 'NEWSLETTER_ERROR';
export const PAYMENT_CARD_ERROR = 'PAYMENT_CARD_ERROR';
export const USER_DETAILS_ERROR = 'USER_DETAILS_ERROR';
export const REGULAR_GUESTS_ERROR = 'REGULAR_GUESTS_ERROR';
export const MEMORABLE_WORD_ERROR = 'MEMORABLE_WORD_ERROR';
export const ROOM_REQUIREMENTS_ERROR = 'ROOM_REQUIREMENTS_ERROR';
export const EXTRAS_PREFERENCES_ERROR = 'EXTRAS_PREFERENCES_ERROR';

const checkForInvalidSession = (error, config, dispatch) => {
  if (InvalidSessionHelper
    .isSessionExpired(error, config)) {
    dispatch({
      type: RESET_USER_INFO,
    });
    dispatch({
      type: RESET_NEWSLETTER_INFO,
    });
    HeaderInterface.userLogOut(HeaderInterface.LOG_OUT_REASONS.SESSION_EXPIRED);
  }
};

export const getSettings = (email, sessionId, business) => (dispatch, getState) => {
  getProfile(email, sessionId, business).then((profile) => {
    dispatch({
      type: PROFILE_RECEIVED,
      profile,
    });
  }).catch((error) => {
    checkForInvalidSession(error, InvalidSessionErrorCodeMap.ACCOUNT.invalidSession, dispatch);
    dispatch({
      type: PROFILE_ERROR,
      error,
    });
  });
  getCountries()
    .then((countries) => {
      dispatch({
        type: COUNTRIES_RECEIVED,
        countries,
      });
    });
  if (getState().application.showEmailPreferences) {
    getNewsletterRegions()
      .then((regions) => {
        dispatch({
          type: REGIONS_RECEIVED,
          regions,
        });
      });

    getNewsletterPreferences(email)
      .then((newsletter) => {
        dispatch({
          type: NEWSLETTER_RECEIVED,
          newsletter,
        });
      })
      .catch(e => dispatch({
        type: NEWSLETTER_ERROR,
        error: e,
      }));
  }
};

export const updateUserProfile = payload => dispatch => updateProfile(payload).catch((error) => {
  checkForInvalidSession(error, InvalidSessionErrorCodeMap.ACCOUNT.invalidSession, dispatch);
  return Promise.reject(error);
});

export const deleteUserProfile = payload => dispatch => deleteCustomer(payload).catch((error) => {
  checkForInvalidSession(error, InvalidSessionErrorCodeMap.DELETE_PROFILE.invalidSession, dispatch);
  return Promise.reject(error);
});

export default {
  getSettings,
};
