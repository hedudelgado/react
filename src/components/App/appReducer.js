import {
  SET_NOTIFICATION_MESSAGE, SET_EDIT_VIEW, DEFAULT_VIEW,
} from './appActions';

const defaultState = {
  editViewName: DEFAULT_VIEW,
};

export default (state = defaultState, action) => {
  switch (action.type) {
    case SET_EDIT_VIEW:
      return {
        ...state,
        editViewName: action.editViewName,
      };
    case SET_NOTIFICATION_MESSAGE:
      return {
        ...state,
        notificationMessage: action.notificationMessage,
      };
    default:
      return state;
  }
};
