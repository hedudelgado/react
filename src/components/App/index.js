export { default } from './App';

export { default as AppActions } from './appActions';

export { default as AppContainer } from './AppContainer';

export { default as modalReducer } from './modalReducer';
