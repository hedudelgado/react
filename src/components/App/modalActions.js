export const MODAL_OPEN = 'MODAL_OPEN';
export const MODAL_CLOSE = 'MODAL_CLOSE';

export const openModal = content => ({
  type: MODAL_OPEN,
  content,
});

export const closeModal = () => ({
  type: MODAL_CLOSE,
});
