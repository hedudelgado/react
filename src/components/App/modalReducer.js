import {
  MODAL_OPEN,
  MODAL_CLOSE,
} from './modalActions';

export default (state = {
  display: false,
  content: null,
}, action) => {
  const {
    type,
    content,
  } = action;

  switch (type) {
    case MODAL_OPEN:
      return {
        ...state,
        display: true,
        content,
      };
    case MODAL_CLOSE:
      return {
        ...state,
        display: false,
        content: null,
      };
    default:
      return state;
  }
};
