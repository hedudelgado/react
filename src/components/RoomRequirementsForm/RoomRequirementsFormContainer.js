import React, { PureComponent } from 'react';
import { func, object } from 'prop-types';
import { Formik } from 'formik';
import BaseStyle from '../../layout/BaseStyle';
import Typography from '../../layout/Typography';
import RoomRequirementsForm from './RoomRequirementsForm';
import {
  ROOM_REQUIREMENTS_VIEW,
  DEFAULT_VIEW,
  ROOM_REQUIREMENTS_ERROR,
} from '../App/appActions';
import Notification from '../../layout/Notification';
import { roomRequirements as roomRequirementsConfig } from '../../config.json';

export default class RoomRequirementsFormContainer extends PureComponent {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(values, actions) {
    const {
      setNotificationMessage,
      updateProfileStore,
      setEditView,
      updateProfile,
    } = this.props;
    const payload = this.createPayload(values);

    updateProfile(payload)
      .then(() => {
        setNotificationMessage(ROOM_REQUIREMENTS_VIEW);
        setEditView(DEFAULT_VIEW);
        actions.setSubmitting(false);
        updateProfileStore(payload);
      }).catch(() => {
        setNotificationMessage(ROOM_REQUIREMENTS_ERROR);
      });
  }

  createPayload(values) {
    const {
      adults,
      children,
      cotRequired,
      type,
    } = values;
    const { profile } = this.props;

    return {
      ...profile,
      bookingPreference: {
        ...profile.bookingPreference,
        roomRequirements: {
          adults: +adults,
          children: +children,
          cotRequired: cotRequired === 'true',
          type,
          hotelBrand: profile.bookingPreference
            && profile.bookingPreference.roomRequirements
            && profile.bookingPreference.roomRequirements.hotelBrand
            ? profile.bookingPreference.roomRequirements.hotelBrand
            : roomRequirementsConfig.defaults.hotelBrand,
        },
      },
    };
  }

  render() {
    const {
      profile,
      setEditView,
      dictionary,
      app: { notificationMessage },
    } = this.props;
    const {
      bookingPreference: {
        roomRequirements: {
          adults = roomRequirementsConfig.defaults.adults,
          children = roomRequirementsConfig.defaults.children,
          cotRequired = roomRequirementsConfig.defaults.cotRequired,
          type = roomRequirementsConfig.defaults.type,
        } = {},
      } = {},
    } = profile;

    return (
      <BaseStyle component="section" className="component-container">
        {notificationMessage === ROOM_REQUIREMENTS_ERROR && (
          <Notification data-test="error-notification" priority={2} variant="error" ariaLive="polite" show>
            {dictionary['generic.error']}
          </Notification>
        )}
        <Typography component="h4" className="pi-color-greyscale-darkest">
          {dictionary['roomrequirements.title.edit']}
        </Typography>

        <Formik
          data-test="room-requirements-form-wrapper"
          onSubmit={this.handleSubmit}
          initialValues={{
            adults,
            children,
            cotRequired,
            type,
          }}
          render={formikProps => (
            <RoomRequirementsForm
              {...formikProps}
              dictionary={dictionary}
              setEditView={setEditView}
              profile={profile}
              roomRequirements={roomRequirementsConfig}
            />
          )}
        />
      </BaseStyle>
    );
  }
}

RoomRequirementsFormContainer.propTypes = {
  app: object.isRequired,
  profile: object.isRequired,
  dictionary: object.isRequired,
  setEditView: func.isRequired,
  setNotificationMessage: func.isRequired,
  updateProfileStore: func.isRequired,
  updateProfile: func.isRequired,
};
