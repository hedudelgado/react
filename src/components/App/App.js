import React, { Component } from 'react';
import {
  array,
  object,
  func,
  bool,
  node,
} from 'prop-types';
import {
  PASSWORD_VIEW,
  PAYMENT_CARD_VIEW,
  USER_DETAILS_VIEW,
  ROOM_REQUIREMENTS_VIEW,
  EXTRAS_PREFERENCES_VIEW,
} from './appActions';
import { ComponentTogglerContainer } from '../ComponentToggler';
import UserProfile from '../UserProfile';
import Password from '../Password';
import UserProfileFormContainer from '../UserProfileForm';
import PasswordFormContainer from '../PasswordForm';
import { PaymentCardContainer } from '../PaymentCard';
import { RegularGuestsContainer } from '../RegularGuests';
import { NewsletterContainer } from '../Newsletter';
import Modal from '../../layout/Modal';
import PaymentCardFormContainer from '../PaymentCardForm';
import { DeleteCustomerContainer } from '../DeleteCustomer';
import RoomRequirements from '../RoomRequirements';
import RoomRequirementsFormContainer from '../RoomRequirementsForm';
import ExtrasPreferences from '../ExtrasPreferences';
import ExtrasPreferencesFormContainer from '../ExtrasPreferencesForm';
import BaseStyle from '../../layout/BaseStyle';
import MemorableWord from '../MemorableWord';
import MoveAccount from '../MoveAccount';
import {
  STAYER, SELF, managePermissionFlag, USER_LOGGED_IN_EVENT,
} from '../../config.json';
import ManagePermission from '../ManagePermission';

class App extends Component {
  constructor(props) {
    super(props);
    this.handleLoggedIn = this.handleLoggedIn.bind(this);
  }

  componentDidMount() {
    window.addEventListener('message', this.handleLoggedIn);
  }

  componentWillUnmount() {
    window.removeEventListener('message', this.handleLoggedIn);
  }

  async handleLoggedIn(event) {
    if (event.data.action === USER_LOGGED_IN_EVENT) {
      const { getSettings } = this.props;
      const { contactDetail: { email }, sessionId, business } = JSON.parse(event.data.profile);
      getSettings(email, sessionId, !!business);
    }
  }

  render() {
    const {
      countries,
      dictionary,
      profile,
      regions,
      newsletter,
      displayModal,
      handleModalClose,
      modalContent,
      userDataReceived,
      application: {
        showEmailPreferences,
      },
    } = this.props;

    const {
      business,
      business: {
        accessLevel = '',
        tetheredGuid = '',
      } = {},
    } = profile;

    const appReady = userDataReceived
      && (!showEmailPreferences || !!regions.length)
      && !!countries.length
      && (!showEmailPreferences || !!Object.keys(newsletter).length)
      && !!Object.keys(dictionary).length
      && !!Object.keys(profile).length;

    const showExtrasPreferences = !(accessLevel === SELF || accessLevel === STAYER);

    return (
      <div data-test="AccountSettings">
        {appReady
        && (
          <BaseStyle className="wb-typography app-container">
            <ComponentTogglerContainer
              viewName={USER_DETAILS_VIEW}
              view={UserProfile}
              edit={UserProfileFormContainer}
            />
            <ComponentTogglerContainer
              viewName={PASSWORD_VIEW}
              view={Password}
              edit={PasswordFormContainer}
            />
            <ComponentTogglerContainer
              viewName={PAYMENT_CARD_VIEW}
              view={PaymentCardContainer}
              edit={PaymentCardFormContainer}
            />
            {managePermissionFlag && (
              <ManagePermission dictionary={dictionary} />
            )}
            <RegularGuestsContainer />
            {showEmailPreferences && (
              <NewsletterContainer />
            )}
            <ComponentTogglerContainer
              viewName={ROOM_REQUIREMENTS_VIEW}
              view={RoomRequirements}
              edit={RoomRequirementsFormContainer}
            />
            {showExtrasPreferences && (
              <ComponentTogglerContainer
                viewName={EXTRAS_PREFERENCES_VIEW}
                view={ExtrasPreferences}
                edit={ExtrasPreferencesFormContainer}
              />
            )}
            {!!tetheredGuid && (
              <MemorableWord />
            )}
            {business && !tetheredGuid && (
              <MoveAccount />
            )}
            {!business && (
            <DeleteCustomerContainer />
            )}
            {displayModal && <Modal data-test="Modal" display content={modalContent} handleClose={handleModalClose} />}
          </BaseStyle>
        )
      }
      </div>
    );
  }
}

App.propTypes = {
  countries: array.isRequired,
  dictionary: object.isRequired,
  profile: object.isRequired,
  getSettings: func.isRequired,
  displayModal: bool.isRequired,
  handleModalClose: func.isRequired,
  modalContent: node,
  newsletter: object.isRequired,
  regions: array.isRequired,
  userDataReceived: bool.isRequired,
  application: object.isRequired,
};

export default App;
