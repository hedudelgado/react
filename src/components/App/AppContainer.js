import { connect } from 'react-redux';
import { getSettings as getSettingsAction } from './appActions';
import { closeModal } from './modalActions';

import App from './App';

const mapStateToProps = ({
  modal: {
    display: displayModal,
    content: modalContent,
  },
  dictionary,
  config,
  countries,
  regions,
  newsletter,
  application,
  userProfile: {
    userDataReceived,
    profile,
  },
}) => ({
  profile,
  displayModal,
  modalContent,
  dictionary,
  config,
  countries,
  regions,
  newsletter,
  application,
  userDataReceived,
});

const mapDispatchToProps = dispatch => ({
  getSettings:
    (email, sessionId, business) => dispatch(getSettingsAction(email, sessionId, business)),
  handleModalClose: () => dispatch(closeModal()),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
