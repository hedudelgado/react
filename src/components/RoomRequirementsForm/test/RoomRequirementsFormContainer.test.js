import React from 'react';
import { shallow } from 'enzyme';
import RoomRequirementsFormContainer from '../RoomRequirementsFormContainer';
import { ROOM_REQUIREMENTS_ERROR } from '../../App/appActions';
import { roomRequirements as roomRequirementsConfig } from '../../../config.json';

describe('RoomRequirementsFormContainer Component', () => {
  const updateProfile = jest.fn(() => Promise.resolve({}));
  let wrapper;
  let props;
  let baseProfile;

  beforeEach(() => {
    baseProfile = {
      businessUse: false,
      companyId: '',
      contactDetail: {
        title: 'Mr',
        firstName: 'Test',
        lastName: 'Test',
        email: 'test@whitbread.com',
        telephone: '111111111',
        mobile: '+351222222222',
        nationality: 'GB',
        passport: {
          number: '',
          countryOfIssue: '',
        },
        carRegistration: '',
        address: {
          line1: '120 Holborn',
          line2: '',
          line3: '',
          line4: 'LONDON',
          line5: '',
          postCode: 'EC1N 2TD',
          countryCode: 'GB',
          type: 'BUSINESS',
          companyName: 'Whitbread',
        },
      },
      guestHistoryNumber: 'G14187551',
      sessionId: 'zdiKE1MflNUmdXAS',
    };
    props = {
      updateProfile,
      app: {},
      profile: {
        ...baseProfile,
        paymentPreference: {
          electronicInvoiceRequired: true,
          paymentCard: {
            cardType: 'VI',
            cardNumber: '************1111',
            expiryDate: '07/23',
            cardHolderName: 'dwa',
          },
        },
        additionalGuests: [
          {
            title: 'Mr',
            firstName: 'Mac',
            lastName: 'Donalds',
            email: 'mac@test.com',
            telephone: '1111111111',
            mobile: '+441111111111',
            nationality: 'RP',
            carRegistration: '',
          },
        ],
        bookingPreference: {
          roomRequirements: {
            type: 'FAM',
            adults: 2,
            children: 1,
            cotRequired: false,
            hotelBrand: 'ZIP',
          },
          foodPreference: 11,
          wantSmsConfirmations: false,
        },
        companyName: 'Pareto Law',
      },
      dictionary: {
        'roomrequirements.title.edit': 'Edit room requirements',
        'roomrequirements.type.double': 'Double',
        'roomrequirements.type.single': 'Single',
        'roomrequirements.type.accessible': 'Accessible',
        'roomrequirements.type.family': 'Family',
      },
      setEditView: jest.fn(),
      setNotificationMessage: jest.fn(),
      updateProfileStore: jest.fn(),
      setFieldValue: jest.fn(),
    };
    wrapper = shallow(<RoomRequirementsFormContainer {...props} />);
  });

  describe('Renders RoomRequirementsFormContainer Element', () => {
    it('should match snapshot', () => {
      expect(wrapper.find('Formik').dive()).toMatchSnapshot();
    });

    it('should show error notification', () => {
      props.app.notificationMessage = ROOM_REQUIREMENTS_ERROR;
      wrapper = shallow(<RoomRequirementsFormContainer {...props} />);
      expect(wrapper).toContainExactlyOneMatchingElement('[data-test="error-notification"]');
    });
  });

  describe('Room Requirements Form submission', () => {
    let setSubmitting;
    let createPayload;

    beforeEach(() => {
      setSubmitting = jest.fn();
      createPayload = jest.fn();
      jest.clearAllMocks();
    });

    it('should execute Profile update success', async () => {
      wrapper.instance().handleSubmit.apply({ props, createPayload }, [
        {},
        { setSubmitting },
      ]);
      await Promise.all([updateProfile]);

      expect(props.setNotificationMessage).toHaveBeenCalledTimes(1);
      expect(props.setEditView).toHaveBeenCalledTimes(1);
      expect(props.updateProfileStore).toHaveBeenCalledTimes(1);
    });

    it('should set default values for missing roomRequirements values', async () => {
      props.profile = baseProfile;
      wrapper = shallow(<RoomRequirementsFormContainer {...props} />);

      expect(wrapper.find('Formik').props().initialValues).toEqual({ ...roomRequirementsConfig.defaults, hotelBrand: undefined });
    });

    it('should set default values for partial missing roomRequirements values and preserve those already set', async () => {
      const partialRoomRequirements = { adults: 2 };

      props.profile.bookingPreference.roomRequirements = partialRoomRequirements;
      wrapper = shallow(<RoomRequirementsFormContainer {...props} />);

      expect(wrapper.find('Formik').props().initialValues).toEqual({ ...roomRequirementsConfig.defaults, ...partialRoomRequirements, hotelBrand: undefined });
    });

    it('should set error notification', async () => {
      updateProfile.mockImplementationOnce(jest.fn(() => Promise.reject({})));
      wrapper.instance().handleSubmit.apply({ props, createPayload }, [{}, {}]);
      await Promise.all([updateProfile]);

      expect(props.setNotificationMessage)
        .toHaveBeenCalledTimes(1)
        .toHaveBeenCalledWith(ROOM_REQUIREMENTS_ERROR);
      expect(props.updateProfileStore).not.toHaveBeenCalled();
    });
  });

  describe('createPayload()', () => {
    const roomRequirements = {
      adults: '1',
      children: '1',
      cotRequired: 'true',
      type: 'FAM',
    };
    it('should set cotRequired to correct boolean value', () => {
      const {
        bookingPreference: {
          roomRequirements: { cotRequired: cotRequiredTrue },
        },
      } = wrapper.instance().createPayload.apply({ props }, [roomRequirements]);

      expect(cotRequiredTrue).toBe(true);

      roomRequirements.cotRequired = 'false';

      const {
        bookingPreference: {
          roomRequirements: { cotRequired: cotRequiredFalse },
        },
      } = wrapper.instance().createPayload.apply({ props }, [roomRequirements]);

      expect(cotRequiredFalse).toBe(false);
    });

    it('should use the profile hotelBrand if set', () => {
      const {
        bookingPreference: {
          roomRequirements: { hotelBrand },
        },
      } = wrapper.instance().createPayload.apply({ props }, [roomRequirements]);
      expect(hotelBrand).toBe('ZIP');
    });

    it('should use the config hotelBrand if profile hotelBrand not set', () => {
      const {
        bookingPreference: {
          roomRequirements: { hotelBrand },
        },
      } = wrapper.instance().createPayload
        .apply({
          props: {
            ...props,
            profile: { ...props.profile, bookingPreference: undefined },
          },
        }, [roomRequirements]);
      expect(hotelBrand).toBe('PI');
    });
  });
});
