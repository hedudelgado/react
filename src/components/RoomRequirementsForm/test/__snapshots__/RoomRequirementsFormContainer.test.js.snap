// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomRequirementsFormContainer Component Renders RoomRequirementsFormContainer Element should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "dirty": false,
      "errors": Object {},
      "handleBlur": [Function],
      "handleChange": [Function],
      "handleReset": [Function],
      "handleSubmit": [Function],
      "initialValues": Object {
        "adults": 2,
        "children": 1,
        "cotRequired": false,
        "type": "FAM",
      },
      "isSubmitting": false,
      "isValid": false,
      "isValidating": false,
      "registerField": [Function],
      "resetForm": [Function],
      "setError": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFormikState": [Function],
      "setStatus": [Function],
      "setSubmitting": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "status": undefined,
      "submitCount": 0,
      "submitForm": [Function],
      "touched": Object {},
      "unregisterField": [Function],
      "validate": undefined,
      "validateField": [Function],
      "validateForm": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
      "validationSchema": undefined,
      "values": Object {
        "adults": 2,
        "children": 1,
        "cotRequired": false,
        "type": "FAM",
      },
    }
  }
>
  <RoomRequirementsForm
    dictionary={
      Object {
        "roomrequirements.title.edit": "Edit room requirements",
        "roomrequirements.type.accessible": "Accessible",
        "roomrequirements.type.double": "Double",
        "roomrequirements.type.family": "Family",
        "roomrequirements.type.single": "Single",
      }
    }
    dirty={false}
    errors={Object {}}
    handleBlur={[Function]}
    handleChange={[Function]}
    handleReset={[Function]}
    handleSubmit={[Function]}
    initialValues={
      Object {
        "adults": 2,
        "children": 1,
        "cotRequired": false,
        "type": "FAM",
      }
    }
    isSubmitting={false}
    isValid={false}
    isValidating={false}
    profile={
      Object {
        "additionalGuests": Array [
          Object {
            "carRegistration": "",
            "email": "mac@test.com",
            "firstName": "Mac",
            "lastName": "Donalds",
            "mobile": "+441111111111",
            "nationality": "RP",
            "telephone": "1111111111",
            "title": "Mr",
          },
        ],
        "bookingPreference": Object {
          "foodPreference": 11,
          "roomRequirements": Object {
            "adults": 2,
            "children": 1,
            "cotRequired": false,
            "hotelBrand": "ZIP",
            "type": "FAM",
          },
          "wantSmsConfirmations": false,
        },
        "businessUse": false,
        "companyId": "",
        "companyName": "Pareto Law",
        "contactDetail": Object {
          "address": Object {
            "companyName": "Whitbread",
            "countryCode": "GB",
            "line1": "120 Holborn",
            "line2": "",
            "line3": "",
            "line4": "LONDON",
            "line5": "",
            "postCode": "EC1N 2TD",
            "type": "BUSINESS",
          },
          "carRegistration": "",
          "email": "test@whitbread.com",
          "firstName": "Test",
          "lastName": "Test",
          "mobile": "+351222222222",
          "nationality": "GB",
          "passport": Object {
            "countryOfIssue": "",
            "number": "",
          },
          "telephone": "111111111",
          "title": "Mr",
        },
        "guestHistoryNumber": "G14187551",
        "paymentPreference": Object {
          "electronicInvoiceRequired": true,
          "paymentCard": Object {
            "cardHolderName": "dwa",
            "cardNumber": "************1111",
            "cardType": "VI",
            "expiryDate": "07/23",
          },
        },
        "sessionId": "zdiKE1MflNUmdXAS",
      }
    }
    registerField={[Function]}
    resetForm={[Function]}
    roomRequirements={
      Object {
        "adults": Array [
          1,
          2,
        ],
        "children": Array [
          0,
          1,
          2,
        ],
        "defaults": Object {
          "adults": 1,
          "children": 0,
          "cotRequired": false,
          "hotelBrand": "PI",
          "type": "DB",
        },
        "type": Array [
          Object {
            "code": "DB",
            "dictionaryKey": "roomrequirements.type.double",
          },
          Object {
            "code": "SB",
            "dictionaryKey": "roomrequirements.type.single",
          },
          Object {
            "code": "TWIN",
            "dictionaryKey": "roomrequirements.type.twin",
          },
          Object {
            "code": "DIS",
            "dictionaryKey": "roomrequirements.type.accessible",
          },
          Object {
            "code": "FAM",
            "dictionaryKey": "roomrequirements.type.family",
          },
        ],
      }
    }
    setEditView={[MockFunction]}
    setError={[Function]}
    setErrors={[Function]}
    setFieldError={[Function]}
    setFieldTouched={[Function]}
    setFieldValue={[Function]}
    setFormikState={[Function]}
    setStatus={[Function]}
    setSubmitting={[Function]}
    setTouched={[Function]}
    setValues={[Function]}
    submitCount={0}
    submitForm={[Function]}
    touched={Object {}}
    unregisterField={[Function]}
    validateField={[Function]}
    validateForm={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    values={
      Object {
        "adults": 2,
        "children": 1,
        "cotRequired": false,
        "type": "FAM",
      }
    }
  />
</ContextProvider>
`;
